### DISCUSSION TITLE
Ability to fetch login-url

### DISCUSSION THREAD

**Discussion by tolsadus** [ThioJoe/YT-Spammer-Purge#491] [Ideas]
When running from a machine without a X Window System, it would be a nice addition to have the oauth url printed to copy/paste on an actual machine to login.

```
Loading YT Spammer Purge @ 2.14.2...
Please login using the browser window that opened just now.
Waiting for authorization. See message above.
```

**Comment by KendallDoesCoding** [comment]
> 

No window opens on a X Window System?

**Reply by tolsadus** [reply]
Yes. I'm currently running this project on a Debian without any GUI. (still looking to keep it running 24/7 on that dedicated machine)

**Reply by KendallDoesCoding** [reply]
Ok, but I dont know if its possible to fetch login-url because google just opens the link in the browser. Also, I dont think @ThioJoe uses Debian so he may not know how to do this. If you know how to do this, feel free to make a PR

**Reply by tolsadus** [reply]
This ain't related to Debian specifically. Yeah i'm looking to fetch the login-url so it can be pasted on another machine. Not 100% sure how to do that, but I've seen this in the past. Will try to find another project that does it?

**Reply by tolsadus** [reply]
https://github.com/GAM-team/GAM is probably the closest I can think about of what I'm looking for.

**Reply by KendallDoesCoding** [reply]
> https://github.com/GAM-team/GAM is probably the closest I can think about of what I'm looking for.

Ok, let @ThioJoe have a look and consider doing this. I mean, we haven't got this as a idea by anyone else, means nobody else wants to run it on another machine, so if its gonna be like a lot of lines of code, I dont think we should add this tbh, but if its not gonna take too much time or lines in the code, ig its a good idea.

**Comment by KendallDoesCoding** [comment]
Ok cool, so what you what exactly is

a) get a url to copy
b) paste it on a windows/linux or mac machine (not using a VM)

If you wanna do this though, I dont think it will work if I'm being honest,
coz Google then redirects you back to the file saying authentication
complete, and it may not work if you paste the url on another machine, but
I'm not really sure.

**Comment by Kurdakov** [comment]
that ability is restricted from google/youtube side,
see this link for details
https://developers.google.com/youtube/v3/guides/auth/server-side-web-apps it's only available for content owners with multiple channels (with no directions how that possibility could be enabled)

**Comment by Dobby233Liu** [comment]
> printed to copy/paste on an actual machine to login.

Printing the url is possible, old versions of this app did that. To achieve this change `Scripts/auth.py` line 74 to
```python
      creds = flow.run_local_server(port=0, authorization_prompt_message="{url}", open_browser=False)
```
However the `localhost` or `127.0.0.1` if there is one in the actual url may need to be changed to the actual ip

**Reply by KendallDoesCoding** [reply]
> 

Please make a PR, and mention this discussion @ThioJoe may consider adding it back.

**Reply by Dobby233Liu** [reply]
Maybe, and I'd put it behind a config option

**Reply by KendallDoesCoding** [reply]
Thanks

**Reply by Kurdakov** [reply]
that approach will work only if he logins on the same machine with say Links2 console browser (if it at all works - but that will be interesting to test and if it does - it will allow to auth via putty on remove web server in future). 

But initial question was  to login somewhere else and it's also possible with server auth path - but see above for youtube it's difficult for user.

**Reply by ThioJoe** [reply]
@KendallDoesCoding Not a good idea to suggest people make a pull request if I haven't even commented on it

**Reply by Firecul** [reply]
> I'd put it behind a config option

I think that's best. Not many people will do headless runs I'd guess

**Reply by KendallDoesCoding** [reply]
> @KendallDoesCoding Not a good idea to suggest people make a pull request if I haven't even commented on it

Sorry
